packages/next/src/server/next.ts

```diff
@@ -197,9 +198,8 @@ export class NextServer implements NextWrapperServer {
   }
 
   async close() {
-    if (this.server) {
-      // BaseServer.close() is protected
-      await this.server['close']()
+    if (this.server && !this.options.dev) {
```

Comment:
Is there a reason this dev check is here? Seems odd that we don't call close in dev

Reply:
removed the `!isDev` special case
https://github.com/vercel/next.js/blob/626b7f9de94a82e0721df9e9a29a5af850dc8ace/packages/next/src/cli/next-dev.ts#L85

my original thinking behind this was to make custom servers mirror the behavior of `next dev` where we exit immediately...

but that fast exit only happens because `next-dev.ts` simply SIGKILLs the process after 100ms: https://github.com/vercel/next.js/blob/626b7f9de94a82e0721df9e9a29a5af850dc8ace/packages/next/src/cli/next-dev.ts#L85
there's nothing there about specifically skipping cleanups. so it's cleaner to make the cleanup do what it should, and if someone wants to avoid waiting for it in a custom server, that's their business